### DISCUSSION TITLE
How to model continuum robots with general tendon routing?

### DISCUSSION THREAD

**Discussion by ZhanchiWang** [GazzolaLab/PyElastica#263] [Q&A] [answered]
Hi there,

I am currently working on a tendon-driven continuum robot project. I'm searching for tools for the modeling and simulation of the robots and have found PyElastica to be a suitable fit.

I have gone through the document to find a suitable _Forcing_ function to simulate the tendon actuation. I have tried MuscleTorques class, but it looks not the same as the cable is pulling.

Can I simulate tendon-driven continuum robots using this software? Is there any _Built-in External Forces_ class suitable for general tendon routing? 

Thanks in advance for your help!

**Comment by armantekinalp** [comment, marked as answer]
Hi @ZhanchiWang ,

You can model tendon-driven robots in PyElastica. However, we do not have any built-in external forces class. You need to write a custom forcing class for that. 

If you are planning to model tendons and backbone as separate rods, you can use [parallel joint](https://github.com/GazzolaLab/PyElastica/blob/master/elastica/experimental/connection_contact_joint/parallel_connection.py) class to connect them.

Let me know if this helps

**Reply by ZhanchiWang** [reply]
Hi @armantekinalp,

Thank you so much for your reply! I will try to work on it under your direction. Will let you know if I work it out!
